pkg/base-dev/lib/story.hoon

```diff
@@ -38,41 +38,17 @@
 ::
 ::  Canonical textual representation
 ::
-++  tako-to-text
-  |=  [=tako:clay]
-  ^-  tape
-  "commit: {<`@uv`tako>}\0a"
-::
-++  proses-to-text
-  |=  [=proses]
-  ^-  tape
-  =/  proses-list=(list prose)  ~(tap in proses)
-  ?:  ?=(~ proses-list)  ""
-  ?:  ?=([prose ~] proses-list)
-    (prose-to-text i.proses-list)
-  %-  tail
-  %^  spin  `(list prose)`t.proses-list
-    (prose-to-text i.proses-list)
-  |=  [prz=prose state=tape]
-  ^-  [prose tape]
-  :-  prz
-  ;:  welp
-    state
-    "|||"
-    "\0a"
-    (prose-to-text prz)
-  ==
-::
-++  prose-to-text
-  |=  prz=prose
-  =/  [title=@t body=@t]  prz
-  ^-  tape
-  ;:  welp
-    "{(trip title)}"
-    "\0a\0a"
-    "{(trip body)}"
-    "\0a"
-  ==
+++  chapter-to-text
+  |=  [=tako:clay =proses]
+  ^-  wain
+  :-  (crip "commit: {<`@uv`tako>}")
+  %-  zing
+  %+  join  `wain`~['|||']
+  %+  turn  ~(tap in proses)
+  |=  prose
+  ^-  wain
+  %-  to-wain:format
+  :((cury cat 3) title '\0a\0a' body)
```

Comment:
For efficiency reasons, `(rap 3 a b c ~)` is always preferable to `:((cury cat 3) a b c)` -- the former is a single call (and a single allocation inside the jet), the latter is `n-1` calls and allocations. (I'm on a personal mission to broadcast this far and wide.)

Reply:
whoops for some reason I totally forgot about rap